### PR TITLE
MINOR: update Kafka Streams docs with 3.2 KIP information

### PR DESCRIPTION
**Pull request by mjsax**

### CODE REVIEW

**Review by cadonna — APPROVED** [review]
Thanks @mjsax !

LGTM!

File: docs/streams/upgrade-guide.html [hunk 1]
```diff
@@ -303,6 +303,61 @@ <h3><a id="streams_api_changes_350" href="#streams_api_changes_350">Streams API
...
+    <h3><a id="streams_api_changes_320" href="#streams_api_changes_320">Streams API changes in 3.2.0</a></h3>
+    <p>
+        RocksDB offers many metrics which are critical to monitor and tune its performance. Kafka Streams started to make RocksDB metrics accessible
+        like any other Kafka metric via <a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-471%3A+Expose+RocksDB+Metrics+in+Kafka+Streams">KIP-471</a> in 2.4.0 release.
+        However, the KIP was only partially implemented, and is now completed with the 3.2.0 release.
+        For a full list of available RocksDB metrics, please consult the <a href="/{{version}}/documentation/#kafka_streams_client_monitoring">monitoring documentation</a>.
+    </p>
+
+    <p>
+        Kafka Streams ships with RocksDB and in-memory store implementations and user can pick which one to use.
```
Comment (JimGalasyn) [inline docs/streams/upgrade-guide.html]:
```suggestion
        Kafka Streams ships with RocksDB and in-memory store implementations and users can pick which one to use.
```

File: docs/streams/upgrade-guide.html [hunk 2]
```diff
@@ -303,6 +303,61 @@ <h3><a id="streams_api_changes_350" href="#streams_api_changes_350">Streams API
...
+        However, for the DSL, the choice is a per-operator one, making it cumbersome to switch from the default RocksDB
+        store to in-memory store for all operators, especially for larger topologies.
+        <a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-591%3A+Add+Kafka+Streams+config+to+set+default+state+store">KIP-591</a>
+        adds a new config <code>default.dsl.store</code> that allows to set the default store for all DSL operators globally.
```
Comment (JimGalasyn) [inline docs/streams/upgrade-guide.html]:
```suggestion
        adds a new config <code>default.dsl.store</code> that enables setting the default store for all DSL operators globally.
```

File: docs/streams/upgrade-guide.html [hunk 3]
```diff
@@ -303,6 +303,61 @@ <h3><a id="streams_api_changes_350" href="#streams_api_changes_350">Streams API
...
+        Note, that it is required to pass <code>TopologyConfig</code> to the <code>StreamsBuilder</code> constructor to make use of this new config.
```
Comment (JimGalasyn) [inline docs/streams/upgrade-guide.html]:
```suggestion
        Note that it is required to pass <code>TopologyConfig</code> to the <code>StreamsBuilder</code> constructor to make use of this new config.
```

File: docs/streams/upgrade-guide.html [hunk 4]
```diff
@@ -303,6 +303,61 @@ <h3><a id="streams_api_changes_350" href="#streams_api_changes_350">Streams API
...
+    </p>
+
+    <p>
+        For multi-AZ deployments, it is desired to assign StandbyTasks to a KafkaStreams instance running in a different
+        AZ than the corresponding active StreamTask.
+        <a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-708%3A+Rack+aware+StandbyTask+assignment+for+Kafka+Streams">KIP-708</a>
+        allows to configure Kafka Streams instances with a rack-aware StandbyTask assignment strategy, by using the new added configs
```
Comment (JimGalasyn) [inline docs/streams/upgrade-guide.html]:
```suggestion
        enables configuring Kafka Streams instances with a rack-aware StandbyTask assignment strategy, by using the new added configs
```

File: docs/streams/upgrade-guide.html [hunk 5]
```diff
@@ -303,6 +303,61 @@ <h3><a id="streams_api_changes_350" href="#streams_api_changes_350">Streams API
...
+
+        In addition, multiple built-in query types were added: <code>KeyQuery</code> for key lookups and
+        <code>RangeQuery</code> (via <a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-805%3A+Add+range+and+scan+query+over+kv-store+in+IQv2">KIP-805</a>)
+        for key-range queries on key-value stores, as well as <code>WindowKeyQuery</code> and <code>WindowRangeQuery</code>
+        (via <a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-806%3A+Add+session+and+window+query+over+kv-store+in+IQv2">KIP-806</a>)
+        for key and range lookup into windowed stores.
+    </p>
+
+    <p>
+        The Kafka Streams DSL may insert so-called repartition topic for certain DSL operators to ensure correct partitioning
```
Comment (JimGalasyn) [inline docs/streams/upgrade-guide.html]:
```suggestion
        The Kafka Streams DSL may insert so-called repartition topics for certain DSL operators to ensure correct partitioning
```

File: docs/streams/upgrade-guide.html [hunk 6]
```diff
@@ -303,6 +303,61 @@ <h3><a id="streams_api_changes_350" href="#streams_api_changes_350">Streams API
...
+        of data. These topics are configured with infinite retention time and Kafka Streams purges old data explicitly
```
Comment (JimGalasyn) [inline docs/streams/upgrade-guide.html]:
```suggestion
        of data. These topics are configured with infinite retention time, and Kafka Streams purges old data explicitly
```

File: docs/streams/upgrade-guide.html [hunk 7]
```diff
@@ -303,6 +303,61 @@ <h3><a id="streams_api_changes_350" href="#streams_api_changes_350">Streams API
...
+        via "delete record" requests, when commiting input topic offsets.
+        <a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-811%3A+Add+config+repartition.purge.interval.ms+to+Kafka+Streams">KIP-811</a>
+        adds a new config <code>repartition.purge.interval.ms</code> allowing to configure the purge interval independently of the commit interval.
```
Comment (JimGalasyn) [inline docs/streams/upgrade-guide.html]:
```suggestion
        adds a new config <code>repartition.purge.interval.ms</code> allowing you to configure the purge interval independently of the commit interval.
```

**Review by JimGalasyn — APPROVED** [review]
LGTM, with a few suggestions.

**Comment by mjsax** [comment]
Merged to `trunk` and cherry-picked to `3.8` and all the way back to `3.2`.

**Pull request merged**